node/src/blockchain/blockchain_interface_utils.rs

```diff
@@ -0,0 +1,1412 @@
+// Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::accountant::db_access_objects::payable_dao::PayableAccount;
+use crate::accountant::db_access_objects::pending_payable_dao::PendingPayable;
+use crate::accountant::scanners::mid_scan_msg_handling::payable_scanner::agent_web3::BlockchainAgentWeb3;
+use crate::accountant::scanners::mid_scan_msg_handling::payable_scanner::blockchain_agent::BlockchainAgent;
+use crate::blockchain::blockchain_bridge::PendingPayableFingerprintSeeds;
+use crate::blockchain::blockchain_interface::blockchain_interface_web3::{
+    to_wei, BlockchainInterfaceWeb3, HashAndAmount, TRANSFER_METHOD_ID,
+};
+use crate::blockchain::blockchain_interface::data_structures::errors::PayableTransactionError;
+use crate::blockchain::blockchain_interface::data_structures::{
+    ProcessedPayableFallible, RpcPayableFailure,
+};
+use crate::sub_lib::blockchain_bridge::ConsumingWalletBalances;
+use crate::sub_lib::wallet::Wallet;
+use actix::Recipient;
+use futures::Future;
+use masq_lib::blockchains::chains::Chain;
+use masq_lib::logger::Logger;
+use secp256k1secrets::SecretKey;
+use serde_json::Value;
+use std::iter::once;
+use std::time::SystemTime;
+use thousands::Separable;
+use web3::transports::{Batch, Http};
+use web3::types::{Bytes, SignedTransaction, TransactionParameters, H256, U256};
+use web3::Error as Web3Error;
+use web3::Web3;
+
```

Comment:
I've already seen some code of this file and understood the philosophy of your actions. I'm not writing this to oppose you, but I just want to record a piece of advice. I can't say from here if it is to work but I'm thinking of limiting the visibility of these functions is really desirable. Therefore the root file of the BlockchainBridge node/src/blockchain/mod.rs should definitely have the line containing a declaration of a bound to this module, that is actually this file, without the keyword `pub`. Why do I care? I just feel a tension in me because of some many public functions potentially contaminating the namespace for more than just the blockchain bridge but even places from much further away from here. 

If you make sure this `mod` is private and so only reachable from the root mod of the BlockchainBridge, then it should leave you more in peace as it would with me, because only the BlockchainBridge will be aware of them.

Not a fatal problem. I just wish you could remember for future occasions.

Reply:
Hi Bert,

Could you please shed some more clarification on this.
The file blockchain_interface_utils.rs is not mark as pub inside mod.rs

Reply:
Good. My apologies, I may've been looking badly at it.  

Anyway, still please remember that this is potentially rather an inadvisable practise. Creating a file with a lot of helper functions, all being public, if it's also coupled with the pub identifier where the individual modules are linked together could bring about namespace pollution and the work becoming slightly less comfortable.